web/src/app/browser/src/oskConfiguration.ts

```diff
@@ -19,16 +19,16 @@ export function setupEmbeddedListeners(osk: OSKView) {
 
   osk.on('hideRequested', (key) => {
     if(osk) {
-      keyman.uiManager.setActivatingUI(false);
+      contextManager.focusAssistant.setMaintainingFocus(false);
       osk.startHide(true);
       keyman.domManager.lastActiveElement = null;
     }
   });
 
-  osk.on('onhide', () => {
+  osk.on('onhide', (hiddenByUser) => {
```

Comment:
This seems like it should be `osk.on('hide'...`?

Reply:
Yeah... there's probably a polish pass to be done somewhere in here.

The point of confusion:

https://github.com/keymanapp/keyman/blob/d5f518205fa4180e606d6db74f4c247f468ec8b2/web/src/engine/osk/src/views/oskView.ts#L43-L53

These are _not_ compatible with `.on` - that's a Node-style event-related method.  They instead use the Web-documented `addEventListener`.  (Note:  Node's events also use `addListener`... but not `addEventListener`.  This is generally not ideal, but at least the related type signatures are distinct and so the two types of events never cross within the same method.)

vs

https://github.com/keymanapp/keyman/blob/d5f518205fa4180e606d6db74f4c247f468ec8b2/web/src/engine/osk/src/views/oskView.ts#L60

https://github.com/keymanapp/keyman/blob/d5f518205fa4180e606d6db74f4c247f468ec8b2/web/src/engine/osk/src/views/oskView.ts#L100-L104

https://github.com/keymanapp/keyman/blob/d5f518205fa4180e606d6db74f4c247f468ec8b2/web/src/engine/osk/src/views/oskView.ts#L215

These use the Node-style signature and are thus compatible with `on`.

We obviously still need to maintain the longtime documented KMW events, and I suppose in my haste to have a shinier, Node-based set for an unattached OSK, I'd written a duplicate set using the Node format.  I should probably just eliminate `onhide` and `onshow` for now and just get by with the `hide` and `show` ones.

Though, for what it's worth, the Node-based ones were far more accessible for unit testing up until the point of the previous PR in this chain - #8591.  Before then, I wasn't sure exactly how the implementation for Web's documented events would end up looking once I got to this stage... back when I was modularizing the OSK.  Now that a pattern for the documented events has been established, the duplicated events here will be much easier to resolve.  

Just... probably not in this PR.  Should I mark this as an issue on the feature-branch for cleanup?

Reply:
> Should I mark this as an issue on the feature-branch for cleanup?

SGTM